### ISSUE TITLE
Apps v2 - (pre-4 and pre-5) `fly.toml` applied to machine but not app

### ISSUE THREAD

**Issue by catflydotio** [superfly/flyctl#1630]
Breaking this out from #1624 ~~since it's independent of `fly launch`.~~ (Actually it's not)

(`fly config save` overwrites `fly.toml` with default `fly.toml` )

To reproduce: 

* deploy an app using a custom `fly.toml` -- say change the internal port and remove health checks
* check that app is reachable at that port
* run `fly config save` and let it overwrite `fly.toml`
* `fly.toml` has `internal_port` 8080 and  the`[[services.tcp_checks]]` stanza is back.

**Comment by catflydotio** [comment]
Oh. `fly.toml` gets applied to the Machine deployed on launch, but not to the App itself.

Here's a flask app listening on 4999.

```
$ fly launch
An existing fly.toml file was found for app jan25
? Would you like to copy its configuration to the new app? Yes
...
Wrote config file fly.toml
...
No machines in jan25 app, launching one new machine
  Machine e784459b655483 update finished: success
  Finished deploying
```
Check the App's config (which is what gets saved if I do `fly config save`):
```
$ fly config show
{
    "env": {},
    "experimental": {
        "auto_rollback": true
    },
    "kill_signal": "SIGINT",
    "kill_timeout": 5,
    "processes": [],
    "services": [
        {
            "concurrency": {
                "hard_limit": 25,
                "soft_limit": 20,
                "type": "connections"
            },
            "http_checks": [],
            "internal_port": 8080,
            "ports": [
                {
                    "force_https": true,
                    "handlers": [
                        "http"
                    ],
                    "port": 80
                },
                {
                    "handlers": [
                        "tls",
                        "http"
                    ],
                    "port": 443
                }
            ],
            "processes": [
                "app"
            ],
            "protocol": "tcp",
            "script_checks": [],
            "tcp_checks": [
                {
                    "grace_period": "1s",
                    "interval": "15s",
                    "restart_limit": 0,
                    "timeout": "2s"
                }
            ]
        }
    ]
}
```

The Machine is OK for the moment, because it did get the config from the local `fly.toml` on launch.

```
$ fly m status e784459b655483 -d
...
Config:
{
  "init": {},
  "image": "registry.fly.io/jan25:deployment-01GQTQTJB5RCS03FKGX9K2H60B",
  "metadata": {
    "fly_platform_version": "v2",
    "fly_release_id": "QgDeJ3o11V2D7fkXVQGvPBON",
    "fly_release_version": "1"
  },
  "restart": {},
  "services": [
    {
      "protocol": "tcp",
      "internal_port": 4999,
      "ports": [
        {
          "port": 80,
          "handlers": [
            "http"
          ],
          "force_https": true
        },
        {
          "port": 443,
          "handlers": [
            "tls",
            "http"
          ]
        }
      ],
      "checks": [
        {
          "type": "tcp",
          "interval": "15s",
          "timeout": "2s"
        }
      ],
      "concurrency": {
        "type": "connections",
        "hard_limit": 25,
        "soft_limit": 20
      }
    }
  ],
  "guest": {
    "cpu_kind": "shared",
    "cpus": 1,
    "memory_mb": 256
  }
}

**Comment by billyb2** [comment]
relevant info:
https://flyio.discourse.team/t/apps-v2-fly-toml-applied-to-machine-but-not-app/1819

**Comment by catflydotio** [comment]
Checked on this with v0.0.474 and the `fly.toml` I got from `fly config save` was correct.

**Comment by billyb2** [comment]
This could've been fixed by #1757
